aptos-move/framework/aptos-stdlib/sources/cryptography/algebra_bls12381.move

```diff
@@ -1,44 +1,765 @@
 module aptos_std::algebra_bls12381 {
     //
-    // Marker types and their serialization schemes begin.
+    // Marker types + serialization formats begin.
     //
 
+    /// The finite field $F_{q^12}$ used in BLS12-381 curves,
+    /// which is an extension field of `Fq6` (defined in the module documentation), constructed as $F_{q^12}=F_{q^6}[w]/(w^2-v)$.
+    struct Fq12 {}
+
+    /// A serialization scheme for `Fq12` elements,
+    /// where an element $(c_0+c_1\cdot w)$ is represented by a byte array `b[]` of size 576,
+    /// which is a concatenation of its coefficients serialized, with the least significant coefficient (LSC) coming first.
+    /// - `b[0..288]` is $c_0$ serialized using `FormatFq6LscLsb` (defined in the module documentation).
+    /// - `b[288..576]` is $c_1$ serialized using `FormatFq6LscLsb`.
+    ///
+    /// NOTE: other implementation(s) using this format: ark-bls12-381-0.4.0.
+    struct FormatFq12LscLsb {}
+
+    /// The group $G_1$ in BLS12-381-based pairing $G_1 \times G_2 \rightarrow G_t$.
+    /// It is a subgroup of `G1Full` (defined in the module documentation) with a prime order $r$
+    /// equal to 0x73eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001.
+    /// (so `Fr` is the associated scalar field).
+    struct G1 {}
+
+    /// A serialization scheme for `G1` elements derived from
+    /// https://www.ietf.org/archive/id/draft-irtf-cfrg-pairing-friendly-curves-11.html#name-zcash-serialization-format-.
+    ///
+    /// Below is the serialization procedure that takes a `G1` element `p` and outputs a byte array of size 96.
+    /// 1. Let `(x,y)` be the coordinates of `p` if `p` is on the curve, or `(0,0)` otherwise.
+    /// 1. Serialize `x` and `y` into `b_x[]` and `b_y[]` respectively using `FormatFqMsb` (defined in the module documentation).
+    /// 1. Concatenate `b_x[]` and `b_y[]` into `b[]`.
+    /// 1. If `p` is the point at infinity, set the infinity bit: `b[0]: = b[0] | 0x40`.
+    /// 1. Return `b[]`.
+    ///
+    /// Below is the deserialization procedure that takes a byte array `b[]` and outputs either a `G1` element or none.
+    /// 1. If the size of `b[]` is not 96, return none.
+    /// 1. Compute the compression flag as `b[0] & 0x80 != 0`.
+    /// 1. If the compression flag is true, return none.
+    /// 1. Compute the infinity flag as `b[0] & 0x40 != 0`.
+    /// 1. If the infinity flag is set, return the point at infinity.
+    /// 1. Deserialize `[b[0] & 0x1f, b[1], ..., b[47]]` to `x` using `FormatFqMsb`. If `x` is none, return none.
+    /// 1. Deserialize `[b[48], ..., b[95]]` to `y` using `FormatFqMsb`. If `y` is none, return none.
+    /// 1. Check if `(x,y)` is on curve `E`. If not, return none.
+    /// 1. Check if `(x,y)` is in the subgroup of order `r`. If not, return none.
+    /// 1. Return `(x,y)`.
+    ///
+    /// NOTE: other implementation(s) using this format: ark-bls12-381-0.4.0.
+    struct FormatG1AffineUncompr {}
```

Comment:
1. For brevity, I would remove `Affine` from here & everywhere else and leave this as `FormatG1Uncompr`. It's unlikely that there will be other formats for BLS12-381.

2. Another alternative to consider is `FormatG1XY` for this and `FormatG1X` (to replace `FormatG1AffineCompr`)

Reply:
Fixed with option 1.